solutions/silver/cf-1088D.mdx

```diff
@@ -0,0 +1,78 @@
+---
+id: cf-1088D
+source: CF
+title: Ehab and another another xor problem
+author: Chongtian Ma
+---
+
+[Official Editorial (C++)](https://codeforces.com/blog/entry/63656)
+
+## Explanation
+
+A good strategy to tackle these problems is just to simulate and try to spot a pattern. Also, keep in mind what matters in each step and construct cases around it. Finally, process queries in a way so that you can differentiate between all of the cases.
+
+Here, we can query from the most significant to the least significant bit. This is because in bitwise, since higher bits always take
+priority, it is hard to bypass the effects of the higher bits if you don't already know them.
+
+First, we can submit a query $(c, d) = (0, 0)$ to figure out whether $a$ or $b$ is greater (this will become helpful later on). Now, loop from most to least significant bits. Let $\mathtt{curA}$ and $\mathtt{curB}$ be the values for the bits of $a$ and $b$ we already know. We can try to differentiate the two numbers at this bit. We can submit two queries, in the form $(\mathtt{curA}| 2^B, \mathtt{curB})$, and $(\mathtt{curA}, \mathtt{curB} | 2^B)$ where we are currently looping on bit $B$. We also know that there are the following four cases we have to consider:
+
+- Both $a$ and $b$ has bit $B$ activated
+- Neither $a$ and $b$ has bit $B$ activated
+- Only $a$ has bit $B$ activated
+- Only $b$ has bit $B$ activated
+
```

Comment:
if you refer to the cases with numbers make this a numbered list

Reply:
unresolved

Reply:
on the live editor it's blue is it supposed to be like that lmao

Reply:
yeah